playerctl: 0.6.1 -> 2.0.1

r-ryantm opened the pull request:
Semi-automatic update generated by https://github.com/ryantm/nixpkgs-update tools. This update was made based on information from https://repology.org/metapackage/playerctl/versions.



meta.description for playerctl is: '"Command-line utility and library for controlling media players that implement MPRIS"'.

[Release on GitHub](https://github.com/acrisci/playerctl/releases/tag/v2.0.1)


[Compare changes on GitHub](https://github.com/acrisci/playerctl/compare/v0.6.1...v2.0.1)

<details>
<summary>
Checks done (click to expand)
</summary>

- built on NixOS
- /nix/store/z1j65p655q7xd4m3bii9fq04cxwdwy1v-playerctl-2.0.1/bin/playerctl passed the binary check.
- 1 of 1 passed binary check by having a zero exit code.
- 1 of 1 passed binary check by having the new version present in output.
- found 2.0.1 with grep in /nix/store/z1j65p655q7xd4m3bii9fq04cxwdwy1v-playerctl-2.0.1
- directory tree listing: https://gist.github.com/51bcd03691cde82388217193d5b69803
- du listing: https://gist.github.com/b93d8e29b67ddaa07d8db00c4e32108a

</details>
<details>
<summary>
Rebuild report (if merged into master) (click to expand)
</summary>

13 total rebuild path(s)

4 package rebuild(s)

4 x86_64-linux rebuild(s)
4 i686-linux rebuild(s)
1 x86_64-darwin rebuild(s)
4 aarch64-linux rebuild(s)


First fifty rebuilds by attrpath
i3blocks
i3lock-pixeled
playerctl
yabar-unstable

</details>

<details>
<summary>
Instructions to test this update (click to expand)
</summary>

Either download from Cachix:
```
nix-store -r /nix/store/z1j65p655q7xd4m3bii9fq04cxwdwy1v-playerctl-2.0.1 \
  --option binary-caches 'https://cache.nixos.org/ https://r-ryantm.cachix.org/' \
  --option trusted-public-keys '
  r-ryantm.cachix.org-1:gkUbLkouDAyvBdpBX0JOdIiD2/DP1ldF3Z3Y6Gqcc4c=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(r-ryantm's Cachix cache is only trusted for this store-path realization.)

Or, build yourself:
```
nix-build -A playerctl https://github.com/r-ryantm/nixpkgs/archive/00f570005752e41a15d34b213a962b0f16bfac1d.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/z1j65p655q7xd4m3bii9fq04cxwdwy1v-playerctl-2.0.1
ls -la /nix/store/z1j65p655q7xd4m3bii9fq04cxwdwy1v-playerctl-2.0.1/bin
```


</details>
<br/>


cc @puffnfresh for testing.

r-ryantm committed:
playerctl: 0.6.1 -> 2.0.1

Semi-automatic update generated by
https://github.com/ryantm/nixpkgs-update tools. This update was made
based on information from
https://repology.org/metapackage/playerctl/versions

jtojnar commented:
gtk-doc should now work: https://github.com/acrisci/playerctl/commit/686cbaa5851fe9d4c6409fa233399e4d3e01977f

worldofpeace committed:
playerctl: enable gtk-doc generation

worldofpeace commented:
> gtk-doc should now work: [acrisci/playerctl@686cbaa](https://github.com/acrisci/playerctl/commit/686cbaa5851fe9d4c6409fa233399e4d3e01977f) cc @hedning

Addressed in https://github.com/NixOS/nixpkgs/pull/52150/commits/59f8d67ec7bd70d6565ecbcf561b6eabf6cadda3

worldofpeace committed:
playerctl: mark as broken on darwin

Build fails with:
../playerctl/playerctl-player.c:274:19: error: use of undeclared identifier 'CLOCK_MONOTONIC'
    clock_gettime(CLOCK_MONOTONIC, &player->priv->cached_position_monotonic);

hedning commented:
Breaks `yabar-unstable`. cc @Ma27 as I noticed you contribute upstream to yabar.

Result of [nix-review](https://github.com/Mic92/nix-review) pr 52150
<details>
  <summary>1 package failed to build:</summary>
  - yabar-unstable
</details>
<details>
  <summary>3 package were build:</summary>
  - i3blocks
  - i3lock-pixeled
  - playerctl
</details>

Ma27 commented:
Thanks a lot for pinging me. I'll try to have a look at this tonight :)

Ma27 commented:
So, I'm afraid that it's not that easy to do the upgrade in the `yabar` sources, also it's questionable when I'll have time to do that.
Until the you could do one the following things:

* supporting multiple versions of playerctl (unless there are security concerns)
* disabling playerctl support in `yabar` (by setting `PLAYERCTL=0` in the build

Mic92 committed:
yabar-unstable: patch for playerctl-2.0 support

Mic92 commented:
third option is to patch yabar to support the new playerctl.

Ma27 commented:
I just had a look at the error last night and was unsure if it was necessary to fix the code itself, not just the build environent.

But thanks a lot for having a deeper look here :+1:
